qiskit_experiments/library/calibration/half_angle_cal.py

```diff
@@ -0,0 +1,147 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Half angle calibration."""
+
+from typing import List, Optional
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.providers.backend import Backend
+
+from qiskit_experiments.framework import ExperimentData
+from qiskit_experiments.calibration_management import (
+    BaseCalibrationExperiment,
+    BackendCalibrations,
+)
+from qiskit_experiments.library.characterization import HalfAngle
+from qiskit_experiments.calibration_management.update_library import BaseUpdater
+
+
+class HalfAngleCal(BaseCalibrationExperiment, HalfAngle):
+    """Calibration version of the half-angle experiment."""
+
+    def __init__(
+        self,
+        qubit,
+        calibrations: BackendCalibrations,
+        backend: Optional[Backend] = None,
+        schedule_name: str = "sx",
+        cal_parameter_name: Optional[str] = "amp",
+        auto_update: bool = True,
+    ):
+        """see class :class:`HalfAngle` for details.
+
+        Args:
+            qubit: The qubit for which to run the half-angle calibration.
+            calibrations: The calibrations instance with the schedules.
+            backend: Optional, the backend to run the experiment on.
+            schedule_name: The name of the schedule to calibrate which defaults to sx.
+            cal_parameter_name: The name of the parameter in the schedule to update. This will
+                default to amp since the complex amplitude contains the phase of the pulse.
+            auto_update:  Whether or not to automatically update the calibrations. By
+                default this variable is set to True.
+        """
+        super().__init__(
+            calibrations,
+            qubit,
+            backend=backend,
+            schedule_name=schedule_name,
+            cal_parameter_name=cal_parameter_name,
+            auto_update=auto_update,
+        )
+
+        self.set_transpile_options(inst_map=calibrations.default_inst_map)
+
+    @classmethod
+    def _default_experiment_options(cls):
+        """Default values for the half angle calibration experiment.
+
+        Experiment Options:
+            result_index (int): The index of the result from which to update the calibrations.
+            group (str): The calibration group to which the parameter belongs. This will default
+                to the value "default".
+
+        """
+        options = super()._default_experiment_options()
+
+        options.result_index = -1
```

Comment:
This can be done later, but I think we should consider searching through results for a name (`d_hac` in this case). Tracking indices does not seem very robust to future changes.

Reply:
I opened an issue to keep track of this. See #478